src/plugins/data/public/ui/query_string_input/_query_bar.scss

```
@@ -55,6 +55,12 @@
   }
 }
 
+.kbnQueryBar__datePickerWrapper {
+  .euiDatePopoverButton-isInvalid {
+    background-image: euiFormControlGradient($euiColorDanger);
+  }
+}
```

I ran these changes by @ryankeairns. It's not a perfect design but it's good enough for now. I opened an EUI issue to get this addressed for real and then we can remove our custom styles.

```suggestion
.kbnQueryBar__datePickerWrapper {
  .euiDatePopoverButton-isInvalid {
    background-image: euiFormControlGradient($euiColorDanger);
  }
  
  // TODO: can remove when EUI issue is resolved
  // https://github.com/elastic/eui/issues/4612
  .euiDatePopoverButton-isInvalid:focus {
    color: $euiTextColor;
    background-color: $euiFormBackgroundColor;
    background-image: euiFormControlGradient($euiColorPrimary);
  }
}
```

---

Done. Thank you!